### ISSUE TITLE
Communicating security-related issues/topics to Jupyter developers

### ISSUE THREAD

**Issue by rcthomas** [jupyter/security#9]
At today's Jupyter community call @sgibson91 mentioned the recent [Travis CI issue](https://arstechnica.com/information-technology/2021/09/travis-ci-flaw-exposed-secrets-for-thousands-of-open-source-projects/) and whether we had a process for alerting Jupyter developers about those kinds of vulnerabilities.  What are the best processes/procedures for getting the word out?  We've got, or could use:

- Discourse (but are security posts visible enough there, and not everyone looks at it)
- Email lists (probably not a lot of excitememt about a new one)
- The Jupyter blog has been used to make statements about security
- Some way to identify Jupyter projects on GitHub and automatically open issues on them (?)

Other ideas?

**Comment by Carreau** [comment]
- We also tweet out when we publish a blog post, or something similar.
 - Dependabot will suggest update when a new version is published.

a suggestion suggestion would be to also have a list of CVE on a given page on Jupyter.org (and rss feeds?).

Im wondering if there is a way to get notified for changes to the github advisory database. Like set a query: https://github.com/advisories?query=jupyter and recieve email when a new entry matches.

**Comment by rpwagner** [comment]
@Carreau I'd like to document the current processes for vulnerability handling. Do you think this is something we could capture during one of the security calls? From there maybe we take it to the Steering Council as a JEP and later refinement.

**Comment by Carreau** [comment]
> I'd like to document the current processes for vulnerability handling. Do you think this is something we could capture during one of the security calls? From there maybe we take it to the Steering Council as a JEP and later refinement.

Yes we can try to capture that, one of the problem is that there is no real process. It's anyone that has access to security@ipython.org does their best.

**Comment by blink1073** [comment]
Here's a sketch of what we've been doing in practice:

- Open a draft [GitHub Security Advisory](https://docs.github.com/en/code-security/repository-security-advisories/about-github-security-advisories-for-repositories#about-github-security-advisories) 
  - Include relevant but sanitized details in the top level comment, which becomes public
  - Sensitive details and reproductions go in the comments on the draft advisory, which are not public
  - Add relevant people to the advisory
  - Decide on a release date and announcement date and post on the advisory.  These are typically at least a week apart to allow administrators to deploy fixes
  - Request a [CVE](https://docs.github.com/en/code-security/repository-security-advisories/about-github-security-advisories-for-repositories#cve-identification-numbers) from GitHub.  This is private until the advisory is published.
- Post the release and announcement dates on the Jupyter Security [Mailing List]([security@ipython.org](mailto:security@ipython.org))
- Create and merge the [vulnerability fix PR](https://docs.github.com/en/code-security/repository-security-advisories/collaborating-in-a-temporary-private-fork-to-resolve-a-repository-security-vulnerability#creating-a-temporary-private-fork)
- Make a release to PyPI and/or npm with no announcement or change log
- Publish the security advisory on the announcement date.  GitHub will post the CVE to the MITRE database

What we haven't yet established is where else to announce these vulnerabilities.

**Comment by blink1073** [comment]
@jweill-aws I've updated the comment above as per our discussion today in the server meeting.
